### PR TITLE
[Merged by Bors] - refactor(LinearAlgebra/ExteriorAlgebra/{Basic,Grading}): add a definition and notation for the exterior powers of a module

### PR DESCRIPTION
**Pull request by smorel394**
This is split off from PR #10654 (that actually proves some properties of the exterior powers). It introduces the reducible definition `exteriorPower R n M` for the `n`th exterior power of the `R`-module `M`; this is of type `Submodule R (ExteriorAlgebra R M)` and defined as `LinearMap.range (ExteriorAlgebra.ι R : M →ₗ[R] ExteriorAlgebra R M) ^ n`. 
It also introduces the notation `Λ[R]^n M` for `exteriorPower R n M`.

Note: for a reason that I don't understand, Lean becomes unable to synthesize the `SetLike.GradedMonoid` instance on `fun (i : ℕ) ↦ (Λ[R]^i) M`, so I added it manually in `ExteriorAlgebra/Graded.lean`. I am far from sure that this is the correct solution.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by mattrobball — APPROVED** [review]
Great to see this! I will wait for @eric-wieser to take a look.

File: Mathlib/LinearAlgebra/ExteriorAlgebra/Basic.lean [hunk 1]
```diff
@@ -66,6 +71,21 @@ def ι : M →ₗ[R] ExteriorAlgebra R M :=
...
 
+section exteriorPower
+
+variable (n : ℕ) (N : Type u2) [AddCommGroup N] [Module R N]
+-- New variables `n` and `N`, to get the correct order of variables in the notation.
+
+/-- Definition of the `n`th exterior power of a `R`-module `N`. We introduce the notation
+`Λ[R]^n N` for `exteriorPower R n N`. -/
+@[reducible]
+def exteriorPower := LinearMap.range (ι R : N →ₗ[R] ExteriorAlgebra R N) ^ n
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/ExteriorAlgebra/Basic.lean:82]: Just to be explicit here
```suggestion
abbrev exteriorPower : Submodule R (ExteriorAlgebra R M) :=
  LinearMap.range (ι R : N →ₗ[R] ExteriorAlgebra R N) ^ n
```
(and `@[reducible] def` is basically the same as `abbrev`)

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

File: Mathlib/LinearAlgebra/ExteriorAlgebra/Basic.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/LinearAlgebra/ExteriorAlgebra/Basic.lean]: (I applied this change)

**Pull request closed without merging**